contrib/src/main/java/com/datatorrent/contrib/enrich/FixedWidthFSLoader.java

```diff
@@ -0,0 +1,248 @@
+ * file.User needs to set {@link FixedWidthFSLoader#fieldDescription} to specify
+ * field information.
+ */
+@InterfaceStability.Evolving
+public class FixedWidthFSLoader extends FSLoader
+{
+
+  private transient List<Field> fields;
+  /**
+   * Indicates whether first line of the file is a header. Default is false
+   */
+  private transient boolean hasHeader;
+
+  /**
+   * Specifies information related to fields in fixed-width file. Format is
+   * [NAME]:[FIELD_TYPE]:[WIDTH]:[date format if FIELD_TYPE is DATE] FIELD_TYPE
+   * can take on of the values of {@link FIELD_TYPE} i.e BOOLEAN, DOUBLE,
+   * INTEGER, FLOAT, LONG, SHORT, CHARACTER, STRING, DATE e.g.
+   * Year:INTEGER:4,Make:STRING:5,Model:STRING:40,Description:STRING:40,
+   * Price:DOUBLE:8,Date:DATE:10:\"dd:mm:yyyy\". Date format needs to be within
+   * quotes (" ")
+   */
+  @NotNull
+  private transient String fieldDescription;
+
+  /**
+   * Array containing headers
+   */
+  private transient String[] header;
+  /**
+   * Array containing width of each field
+   */
+  private transient int[] width;
+  /**
+   * Padding character. Default is white space.
+   */
+  private transient char padding = ' ';
+  private transient FixedWidthParser fixedWidthParser;
+  private transient boolean initialized;
+  /**
+   * Map holding the format for date fields
+   */
+  private transient Map<String, String> fieldDateFormat = Maps.newHashMap();
+
+  private static final Logger logger = LoggerFactory.getLogger(FixedWidthFSLoader.class);
+
+  public FixedWidthFSLoader()
+  {
+  }
+
+  /**
+   * Gets the option if file has header or not.
+   *
+   * @return hasHeader
+   */
+  public boolean isHasHeader()
+  {
+    return hasHeader;
+  }
+
+  /**
+   * Set to true if file has header
+   * 
+   * @param hasHeader
+   */
+  public void setHasHeader(boolean hasHeader)
+  {
+    this.hasHeader = hasHeader;
+  }
+
+  /**
+   * Gets the field description
+   * 
+   * @return fieldDescription
+   */
+  public String getFieldDescription()
+  {
+    return fieldDescription;
+  }
+
+  /**
+   * Sets fieldDescription i.e a String specifying information related to fields
+   * in fixed-width file. Format is [NAME]:[FIELD_TYPE]:[WIDTH]:[date format if
+   * FIELD_TYPE is DATE] FIELD_TYPE can take on of the values of
+   * {@link FIELD_TYPE}
+   * e.g.Year:INTEGER:4,Make:STRING:5,Model:STRING:40,Description:STRING:40,
+   * Price:DOUBLE:8,Date:DATE:10:\"dd:mm:yyyy\" Date format needs to be within
+   * quotes (" ")
+   * 
+   * @param fieldDescription
+   */
+  public void setFieldDescription(String fieldDescription)
+  {
+    this.fieldDescription = fieldDescription;
+  }
+
+  /**
+   * Gets the character used for padding in the fixed-width file.Default is
+   * white space (' ')
+   * 
+   * @return
+   */
+  public char getPadding()
+  {
+    return padding;
+  }
+
+  /**
+   * Sets the character used for padding in fixed-width file.Default is white
+   * space (' ')
+   * 
+   * @param padding
```

Comment:
Please take care of descriptions of parameters and return type in javadoc everywhere else.